src/chessboard/hooks/useArrows.ts

```diff
@@ -25,6 +25,8 @@ export const useArrows = (
         //filter out arrows which starts and ends in the same square
         customArrows?.filter((arrow) => arrow[0] !== arrow[1])
       );
+      // makes it so that custom arrows also controls current arrows drawn on the board
+      setArrows(customArrows?.filter((arrow) => arrow[0] !== arrow[1]));
```

Comment:
I'm not sure if this is behaviour that we want,

We want users to be able to set custom arrows, but then continue to draw additional arrows onto the board on top of them, hence the `allBoardArrows` that is returned from this hook

Reply:
we do currently expose a function to clearPremoves via a ref, maybe a similar function could be exposed

Reply:
> We want users to be able to set custom arrows, but then continue to draw additional arrows onto the board on top of them

This change doesn't interfere with the user's ability to draw arrows on the board whenever they want, but makes it so arrows can be controlled from a single source of truth. The temporary arrows get cleared anyway if you change the position, so why not give the developer control over them programmatically?
A separate property to control the temporary arrows would also do the trick.

Reply:
Apologies for the late reply @Epitomaniac 

Let's think though all the scenarios and how I think this package should respond:

- Users should be able to draw arrows on the board with their mouse whilst holding right click
- Arrows should be able to be set programmatically and display on the board
- If a user has drawn arrows on the board, and a new event fires such as a new position, or a button click external to the board that causes new custom arrows to be sent to it, clear the user drawn arrows and show any new custom arrows
- If custom arrows are drawn on the board, the user should be able to draw additional arrows on the board, alongside the custom arrows (e.g. a position is shown with custom arrows and the user draws their next possible moves)

After writing out those scenarios and re-checking this PR, I'm pretty sure every criteria is met. I'll check in the preview environment and if all is good I'll be happy to merge

Reply:
So I've tested it and found an issue,

If the board re-renders then user arrows will be cleared, you can reproduce this by:
- going to https://react-chessboard-git-for-cf6745-ryan-gregorys-projects-84c47da3.vercel.app/?path=%2Fstory%2Fexample-chessboard--board-with-custom-arrows&vercelToolbarCode=I8LECiyv80Y2eVT 
- drawing a user arrow
- selecting another colour

This is likely because:
- The story re-renders
- This recalls the Chessboard component with the custom arrows props
- Because the custom arrows prop is an array, it is deemed to have changed and fires the useEffect
- This sets the user arrows to match the custom arrows, effectively wiping the user arrows

Reply:
Hello and thanks for the reply. I'll be working on a fix.

Reply:
On a deeper look, I realized that calling setArrows inside the hook is a mistake and would create duplicate arrows on the board (cause custom arrows and user arrows to stack on top of each other). The correct method is to call clearArrows before setting customArrows; this way the customArrows overwrites use arrows without the risk of creating duplicate arrows.
About the bug: I realized this happens only with a ChessBoard with hardcoded customArrows, but if you receive the value from useState, the problem is resolved. It's more likely a React thing rather than a bug.
```
useEffect(() => {
    if (Array.isArray(customArrows)) {
      // so that custom arrows overwrite temporary arrows
      clearArrows();
      setCustomArrows(
        //filter out arrows which starts and ends in the same square
        customArrows?.filter((arrow) => arrow[0] !== arrow[1])
      );
    }
  }, [customArrows]);
```
```
export const BoardWithCustomArrows = () => {
  const colorVariants = [
    "darkred",
    "#48AD7E",
    "rgb(245, 192, 0)",
    "#093A3E",
    "#F75590",
    "#F3752B",
    "#058ED9",
  ];
  const [activeColor, setActiveColor] = useState(colorVariants[0]);
  const [customArrows, setCustomArrows] = useState([
    ["a2", "a3", colorVariants[0]],
    ["b2", "b4", colorVariants[1]],
    ["c2", "c5", colorVariants[2]],
  ]);
  return (
    <div style={boardWrapper}>
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          alignItems: "center",
        }}
      >
        <div>Choose arrow color</div>
        <div
          style={{
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
          }}
        >
          {colorVariants.map((color) => (
            <div
              style={{
                width: "24px",
                height: "24px",
                background: color,
                borderRadius: "50%",
                cursor: "pointer",
                margin: "16px 6px",
                transform: color === activeColor ? "scale(1.5)" : "none",
                transition: "all 0.15s ease-in",
              }}
              onClick={() => setActiveColor(color)}
            />
          ))}
        </div>
      </div>
      <Chessboard
        id="BoardWithCustomArrows"
        customArrows={customArrows}
        customArrowColor={activeColor}
        onArrowsChange={console.log}
      />
    </div>
  );
};
```